ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivGetPrivateTransaction.java

```diff
@@ -40,13 +39,15 @@
   private static final Logger LOG = getLogger();
 
   private final BlockchainQueries blockchain;
-  private final Enclave enclave;
+  private final PrivateTransactionHandler privateTransactionHandler;
 
   public PrivGetPrivateTransaction(
-      final BlockchainQueries blockchain, final PrivacyParameters privacyParameters) {
+      final BlockchainQueries blockchain,
+      final PrivacyParameters privacyParameters,
```

Comment:
this is probably a smell - the "IsPrivacyEnabled" shouldn't be a base-class requirement for every RPC - maybe its a decorator - or more likely, if privacy is disabled, we don't even install the right handlers, we install "NOT_ENABLED"handler instead.

Reply:
Tend to agree with you there, but thinking seperate PR as I think there is bit more to this. Thinking it shouldn't be possible to even enabled the privacy apis if privacy isn't enabled.